extensions/pkg/controller/heartbeat/reconciler.go

```diff
@@ -42,32 +42,27 @@ func NewReconciler(mgr manager.Manager, extensionName string, namespace string,
 // Reconcile renews the heartbeat lease resource.
 func (r *reconciler) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
 	log := logf.FromContext(ctx)
+
 	lease := &coordinationv1.Lease{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      extensions.HeartBeatResourceName,
 			Namespace: r.namespace,
 		},
+		Spec: coordinationv1.LeaseSpec{
+			HolderIdentity:       &r.extensionName,
+			LeaseDurationSeconds: &r.renewIntervalSeconds,
+		},
 	}
 
-	if err := r.client.Get(ctx, client.ObjectKeyFromObject(lease), lease); err != nil {
-		if apierrors.IsNotFound(err) {
-			lease.Spec = coordinationv1.LeaseSpec{
-				HolderIdentity:       &r.extensionName,
-				LeaseDurationSeconds: &r.renewIntervalSeconds,
-				RenewTime:            &metav1.MicroTime{Time: r.clock.Now().UTC()},
-			}
-			log.V(1).Info("Creating heartbeat Lease", "lease", client.ObjectKeyFromObject(lease))
-			return reconcile.Result{RequeueAfter: time.Duration(r.renewIntervalSeconds) * time.Second}, r.client.Create(ctx, lease)
-		}
+	op, err := controllerutil.CreateOrUpdate(ctx, r.client, lease, func() error {
+		lease.Spec.RenewTime = &metav1.MicroTime{Time: r.clock.Now().UTC()}
```

Comment:
I would still put all of the modification of the spec here. If the `r.extensionName` or (more likely) the `r.renewIntervalSeconds` change, they will be overwritten by the existing lease, as `CreateOrUpdate` does a `Get` first